### PR TITLE
PLANNER-1217 FlightCrewSequentialPillarSwapMoveIteratorFactory

### PR DESCRIPTION
**Pull request by ge0ffrey**
For Donato

### CODE REVIEW

**Comment by ge0ffrey** [comment]
jenkins retest this

File: optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMove.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            Employee rightEmployee, List<FlightAssignment> rightFlightAssignmentList) {
+        this.leftEmployee = leftEmployee;
+        this.leftFlightAssignmentList = leftFlightAssignmentList;
+        this.rightEmployee = rightEmployee;
+        this.rightFlightAssignmentList = rightFlightAssignmentList;
+    }
+
+    @Override
+    public boolean isMoveDoable(ScoreDirector<FlightCrewSolution> scoreDirector) {
+        return true;
```
Comment (rsynek) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMove.java]: Should be the move doable is both left and right `FlightAssignmentList` are empty?

File: optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMoveIteratorFactory.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            // it should be possible to select leftEmployee as null and the flightAssignmentSet assigned to null
+            int leftEmployeeIndex = workingRandom.nextInt(employeeList.size());
+            int rightEmployeeIndex = workingRandom.nextInt(employeeList.size() - 1);
+            if (rightEmployeeIndex >= leftEmployeeIndex) {
+                rightEmployeeIndex++;
+            }
+            Employee leftEmployee = employeeList.get(leftEmployeeIndex);
+            Employee rightEmployee = employeeList.get(rightEmployeeIndex);
+            List<FlightAssignment> leftFlightAssignmentList = selectSequentialSubPillar(leftEmployee.getFlightAssignmentSet(), 2);
+            List<FlightAssignment> rightFlightAssignmentList = selectSequentialSubPillar(rightEmployee.getFlightAssignmentSet(), 0);
```
Comment (rsynek) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMoveIteratorFactory.java]: Why passing 0 as a `minimumSize`? The move will degrade into pillar change if one of the lists is empty.

Comment (ge0ffrey) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMoveIteratorFactory.java]: every one of these move is also a pillar change or swap move. The difference is that this factory selects a subset of those: subpillars that have sequential elements.

Comment (rsynek) [reply optaplanner-examples/src/main/java/org/optaplanner/examples/flightcrewscheduling/solver/move/FlightCrewSequentialPillarSwapMoveIteratorFactory.java]: Ok, I was a bit concerned about the fact this is a swap move, so was not sure if the change move should be permitted too.

**Comment by triceo** [comment]
Closing as we have generic sequential moves now.

**Pull request closed without merging**